### PR TITLE
Arctan avx512 ready

### PR DESCRIPTION
**Pull request by Ka-zam**
This is comprehensive PR with added NaN and Inf testing. Can be merged when #798 is accepted.

### CODE REVIEW

File: include/volk/volk_avx_intrinsics.h [hunk 1]
```diff
@@ -24,7 +24,7 @@
  * Maximum relative error ~6.5e-7
  * Polynomial evaluated via Horner's method
  */
-static inline __m256 _m256_arctan_poly_avx(const __m256 x)
+static inline __m256 _mm256_arctan_poly_avx(const __m256 x)
```
Comment (jdemel) [inline include/volk/volk_avx_intrinsics.h:27]: nice catch.

**Review by jdemel — COMMENTED** [review]
Thanks for this PR!

I assume it supersedes #759, correct?

Generally, this PR is fine. Though it looks like it needs a rebase. The RISCV tests broke and GH reports a lot of commits by other people that seem to be part of this PR but shouldn't. 
I assume a rebase would fix it.

**Comment by jdemel** [comment]
I just merged #798 thanks! I assume you can just rebase against latest `main` and then this PR should be ready too.

**Comment by Ka-zam** [comment]
This is quite comperehensive and 23 kernels were touched. 

```
volk_profile \
    -R volk_16i_convert_8i \
    -R volk_16i_s32f_convert_32f \
    -R volk_16ic_convert_32fc \
    -R volk_32f_accumulator_s32f \
    -R volk_32f_acos_32f \
    -R volk_32f_asin_32f \
    -R volk_32f_atan_32f \
    -R volk_32f_cos_32f \
    -R volk_32f_log2_32f \
    -R volk_32f_s32f_convert_16i \
    -R volk_32f_s32f_convert_8i \
    -R volk_32f_sqrt_32f \
    -R volk_32fc_32f_dot_prod_32fc \
    -R volk_32fc_accumulator_s32fc \
    -R volk_32fc_convert_16ic \
    -R volk_32fc_deinterleave_32f_x2 \
    -R volk_32fc_magnitude_32f \
    -R volk_32fc_s32f_atan2_32f \
    -R volk_32fc_s32fc_rotator2puppet_32fc \
    -R volk_32fc_x2_conjugate_dot_prod_32fc \
    -R volk_32fc_x2_divide_32fc \
    -R volk_8i_convert_16i \
    -R volk_8i_s32f_convert_32f

.
.
.

RUN_VOLK_TESTS: volk_8i_s32f_convert_32f(131071,1987)
generic                            8.2697 ms (157471.4 MB/s)
u_avx2                             7.1010 ms (183389.5 MB/s)
u_avx512                           5.7801 ms (225296.2 MB/s) *
u_sse4_1                          14.3626 ms ( 90669.1 MB/s)
a_avx2                             7.2725 ms (179062.5 MB/s)
a_avx512                           5.3665 ms (242662.3 MB/s) *
a_sse4_1                          14.5474 ms ( 89517.2 MB/s)
Best aligned arch:                  a_avx512         (1.54x)
Best unaligned arch:                u_avx512         (1.43x)
--------------------------------------------------------------------------------

Session summary (23 kernels):
  Average speedup: 19.83x
  Max speedup: 105.68x (volk_32f_atan_32f)
```

Also introduced edge cases in tests to make sure atan2 behaves:
```
  const float inf = std::numeric_limits<float>::infinity();
  const float nan = std::nanf("");
  test_params_atan2.add_complex_edge_cases(
      { lv_cmake(0.0f, 0.0f),   // atan2(0, 0) = 0
        lv_cmake(0.0f, -0.0f),  // atan2(-0, 0) = -0 (preserve sign)
        lv_cmake(inf, inf),     // atan2(inf, inf) = π/4
        lv_cmake(nan, 1.0f),    // atan2(1, nan) = nan
        // ... more cases
      });
```
